### PR TITLE
Icelandic shields

### PR DESCRIPTION
**Pull request by riiga**
Adds support for Icelandic shields as per issue #390.

I only added the main network values, not any of the combinations, but this covers 90-something % of all roads. If we could get the Icelandic community to prefix these with `IS:`, full support could be added since all shields look the same. It might also be a good idea to suggest ordering the combinations so that `S;T;L` and `T;S;L` are equivalent and only written as `S;T;L`.

### CODE REVIEW

**Comment by ZeLonewolf** [comment]
Supporting unprefixed single-letter network values seems particularly problematic.

**Comment by claysmalley** [comment]
`Sx` is the same as the country code for the Dutch part of Saint Martin. We cannot add support for these shields until the Icelandic community determines a new `network=*` tagging scheme that does not collide with any other countries.

**Review by ZeLonewolf — CHANGES_REQUESTED** [review]
1. Coordinate with the mapping community in Iceland to define `network` values consistent with international norms
2. Update this PR consistent with those definitions.

File: src/js/shield_defs.js [hunk 1]
```diff
@@ -2761,6 +2761,34 @@ export function loadShields(shieldImages) {
...
+    "H", // Héraðsvegur - County road
+    // Combinations used as of 2022-06-19
+    // Should these be supported?
+    //"S;H",
+    //"S;Sx",
+    //"S;T",
+    //"S;T;L",
+    //"T;H",
+    //"T;L",
+    //"T;S;L",
```
[the remark this answers is not in the document]
Comment (riiga) [reply src/js/shield_defs.js:2779]: From what I can deduce, it's the latter: the classification of the road changes, but it has the same number (and shield). Hence why, if we could just prefix with `IS:`, the rest of the network won't matter as there's only one type of shield in use.

**Comment by ZeLonewolf** [comment]
I've been to Iceland, and in my experience there are only two kinds of routes: roads and F roads.   The network of one to three digit roads, generally with black on white colors, has no on the ground differentiation that I can see. 

Image below, the dashed border means a road to that numbered route. 



The other type, F roads, have the same visuals but are prefixed with F.  They indicate mountain roads and are seasonal and require an off road vehicle. Fords and difficult terrain are typical.

**Comment by 1ec5** [comment]
> Image below, the dashed border means a road to that numbered route.

This would be a third route type, right? Several Scandinavian countries apparently denote link roads with dashed borders.

**Comment by claysmalley** [comment]
According to the [OSM Wiki](https://wiki.openstreetmap.org/wiki/Icelandic_road_system), the current `network=*` tagging scheme in Iceland seems to represent functional classification. This scheme can be seen [here](https://www.vegagerdin.is/vegakerfid/vegaskra/) (click on the image on the right, or the PDF link at the bottom).

Shields in Iceland may all be identical, though it appears many are unsigned. I see many routes with 4-digit numbers but it looks like only routes with 1-3 digits (and maybe an `F` prefix) have signs on the ground.

**Comment by riiga** [comment]
I've sent a mail to the mailing list.
https://lists.openstreetmap.org/pipermail/talk-is/2022-June/001077.html

**Comment by 1ec5** [comment]
Is it even necessary to mention the classification in the `network` value? In other words, when someone refers to the route as a whole, do they call it “Tengivegur/Landsvegur 22”? (I’m making up the number here.) Or is it actually two distinct routes, “Tengivegur 22” and “Landsvegur 22”, that happen to share the same number and connect to one another?

**Comment by ZeLonewolf** [comment]
Seeing as there doesn't seem to be much if any in the way of local engagement, from everything I've read and based on my on the ground experience driving around the country for a week or so, here's my hot take:

There is one, single, numbered national road network in Iceland, and the functional classifications are a separate concept.  I'm inclined to think that the most sensible scheme for route relations for Icelandic roads is:
`network=IS`
`ref=<route number>`

The only thing that comes close to being a separate network are the "F" roads, but they're consistent with the other non-F roads in their numbering scheme and signage.

**Comment by riiga** [comment]
I'm inclined to agree. If no one objects, I will move the current data contained in the network tag to `road_class` and then use `network=IS` for all relations.

**Comment by 1ec5** [comment]
> I'm inclined to agree. If no one objects, I will move the current data contained in the network tag to `road_class` (or any better suggestion) and then use `network=IS` for all relations.

Note that “road class” is sometimes used to describe the `highway` key. There’s precedent for putting official road classifications in [`designation`](https://wiki.openstreetmap.org/wiki/Key:designation), but there are also plenty of examples of countries that have devised country-specific keys for this purpose.

**Comment by riiga** [comment]
Designation seems to be the key I was looking for

**Comment by riiga** [comment]
[Brace for impact](https://www.openstreetmap.org/changeset/123437050)

**Comment by claysmalley** [comment]
Don't forget `npm run status_map` to add Iceland to the shield coverage map.

**Comment by ZeLonewolf** [comment]
That must have been quite the effort @riiga 
image

I will update this PR when I can get around to re-rendering the planet.

**Comment by riiga** [comment]
@ZeLonewolf Overpass-Turbo + JOSM made quick work of it, though I did run out of memory a couple of times in JOSM (remember to disable validator checks on upload when working with large amounts of data).

**Review by 1ec5 — APPROVED** [review]

**Comment by 1ec5** [comment]
[](https://zelonewolf.github.io/openstreetmap-americana/#12.68/64.80019/-21.46465)

**Pull request merged**